crates/spirv-std/src/textures.rs

```diff
@@ -1,10 +1,81 @@
-use glam::{Vec2, Vec3A, Vec4};
+use glam::Vec4;
 
-#[allow(unused_attributes)]
-#[spirv(sampler)]
-#[derive(Copy, Clone)]
-pub struct Sampler {
-    _x: u32,
+use crate::ScalarOrVector;
+
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum Dimensionality {
+    OneD = 0,
+    TwoD = 1,
+    ThreeD = 2,
+    Cube = 3,
+    Rect = 4,
+    Buffer = 5,
+    SubpassData = 6,
+}
+
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum ImageDepth {
+    No = 0,
+    Yes = 1,
+    Unknown = 2,
+}
+
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum Sampled {
+    Unknown = 0,
+    Yes = 1,
+    No = 2,
+}
+
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum ImageFormat {
+    Unknown = 0,
+    Rgba32f = 1,
+    Rgba16f = 2,
+    R32f = 3,
+    Rgba8 = 4,
+    Rgba8Snorm = 5,
+    Rg32f = 6,
+    Rg16f = 7,
+    R11fG11fB10f = 8,
+    R16f = 9,
+    Rgba16 = 10,
+    Rgb10A2 = 11,
+    Rg16 = 12,
+    Rg8 = 13,
+    R16 = 14,
+    R8 = 15,
+    Rgba16Snorm = 16,
+    Rg16Snorm = 17,
+    Rg8Snorm = 18,
+    R16Snorm = 19,
+    R8Snorm = 20,
+    Rgba32i = 21,
+    Rgba16i = 22,
+    Rgba8i = 23,
+    R32i = 24,
+    Rg32i = 25,
+    Rg16i = 26,
+    Rg8i = 27,
+    R16i = 28,
+    R8i = 29,
+    Rgba32ui = 30,
+    Rgba16ui = 31,
+    Rgba8ui = 32,
+    R32ui = 33,
+    Rgb10a2ui = 34,
+    Rg32ui = 35,
+    Rg16ui = 36,
+    Rg8ui = 37,
+    R16ui = 38,
+    R8ui = 39,
+}
+
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum AccessQualifier {
+    ReadOnly = 0,
+    WriteOnly = 1,
+    ReadWrite = 2,
```

Comment:
Rust guarantees sequential values starting from `0`, not sure why they were explicitly provided here.

Reply:
Well these are also the values of their SPIR-V assembly, and including them would allow us to re-order them and not worry about it affecting code, where as implicit sequential values relies on us, rustfmt, or a contributor remembering to not re-order them.